/*
 * University of Illinois Open Source License
 * Copyright 2012-2016 Roberts Group,
 * All rights reserved.
 *
 * Developed by: Roberts Group
 *                  Johns Hopkins University
 *                  http://biophysics.jhu.edu/roberts/
 *
 * Permission is hereby granted, free of charge, to any person obtaining a copy of
 * this software and associated documentation files (the Software), to deal with
 * the Software without restriction, including without limitation the rights to
 * use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies
 * of the Software, and to permit persons to whom the Software is furnished to
 * do so, subject to the following conditions:
 *
 * - Redistributions of source code must retain the above copyright notice,
 * this list of conditions and the following disclaimers.
 *
 * - Redistributions in binary form must reproduce the above copyright notice,
 * this list of conditions and the following disclaimers in the documentation
 * and/or other materials provided with the distribution.
 *
 * - Neither the names of the Roberts Group, Johns Hopkins University,
 * nor the names of its contributors may be used to endorse or
 * promote products derived from this Software without specific prior written
 * permission.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.  IN NO EVENT SHALL
 * THE CONTRIBUTORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR
 * OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
 * ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
 * OTHER DEALINGS WITH THE SOFTWARE.
 *
 * Author(s): Elijah Roberts, Max Klein
 */
syntax = "proto2";
package lm.io.ffpilot;

import "lm/input/ffpilot/FFPilotPhaseLimit.proto";
import "lm/io/ffpilot/FFPilotPhaseOutput.proto";

message FFPilotPhase {
    enum TrajectoryDuplication {
        NONE=0;
        CYCLIC=1;
        UNIFORM_RANDOM=2;
    }

    enum TrajectoryGeneration {
        EAGER=0;
        LAZY=1;
    }

    optional int64 phase_id      = 1;
    optional int64 tile_id       = 2;
    optional int64 basin_id      = 3;
    optional uint64 tiling_id    = 4;
    optional uint64 replicate_id = 5 [default=0];

    optional uint64 first_trajectory_id                   = 11;
    optional uint64 last_trajectory_id                    = 12;
    optional TrajectoryDuplication trajectory_duplication = 13 [default=NONE];
    optional TrajectoryGeneration trajectory_generation   = 14 [default=LAZY];
    optional uint64 simultaneous_trajectories             = 15;

    optional lm.input.ffpilot.FFPilotPhaseLimit ffpilot_phase_limit = 101;

    repeated lm.io.ffpilot.EndPoint start_points                    = 103;

}
